### PR TITLE
DX-2231 Updated to Include Matrix Testing

### PR DESCRIPTION
**Pull request by ckoegel**

### CODE REVIEW

**Comment by hamermike** [comment]
@ckoegel what if I told you there were different Java distributions?
https://github.com/actions/setup-java/blob/main/docs/advanced-usage.md#Testing-against-different-Java-distributions
👀

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -1,21 +1,29 @@
...
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up JDK 11
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: Set up JDK
       uses: actions/setup-java@v2
       with:
         distribution: 'adopt'
```
Comment (hamermike) [inline .github/workflows/test.yml:23]: `adopt` is being phased out and replaced with `temurin`. I know you didn't mess with this but we should update.

Comment (ckoegel) [reply .github/workflows/test.yml:23]: should I make a matrix for the distributions?

Comment (ckoegel) [reply .github/workflows/test.yml:23]: akin to `distribution: [ 'zulu', 'temurin' ]` with `distribution: ${{ matrix.distribution }}`?

**Review by ajrice6713 — DISMISSED** [review]

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -1,21 +1,30 @@
...
-  build:
-    runs-on: ubuntu-latest
+  test:
+    name: Test
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [windows-2016, windows-2019, ubuntu-18.04, ubuntu-20.04]
+        java-version: ['8', '11']
+        distribution: [ 'zulu', 'temurin' ]
```
Comment (hamermike) [inline .github/workflows/test.yml:16]:
```suggestion
        distribution: ['zulu', 'temurin']
```

File: README.md [hunk 3]
```diff
@@ -1,5 +1,13 @@
 # Bandwidth Java SDK
 
+[![Test](https://github.com/Bandwidth/java-sdk/actions/workflows/test.yaml/badge.svg)](https://github.com/Bandwidth/java-sdk/actions/workflows/test.yaml)
+
+| **OS** | **Java** |
+|:---:|:---:|
+| Windows 2016 | 8, 11 |
+| Windows 2019 | 8, 11 |
+| Ubuntu 18.04 | 8, 11 |
+| Ubuntu 20.04 | 8, 11 |
```
Comment (hamermike) [inline README.md:10]:
```suggestion
| **OS** | **Java** | **Distribution** |
|:---:|:---:|:---:|
| Windows 2016 | 8, 11 | Zulu OpenJDK, Eclipse Temurin |
| Windows 2019 | 8, 11 | Zulu OpenJDK, Eclipse Temurin |
| Ubuntu 18.04 | 8, 11 | Zulu OpenJDK, Eclipse Temurin |
| Ubuntu 20.04 | 8, 11 | Zulu OpenJDK, Eclipse Temurin |
```

File: src/test/java/com/bandwidth/ApiTest.java [hunk 4]
```diff
@@ -174,12 +174,12 @@ public void testCreateCallWithAmdAndGetCallState() throws Exception {
...
-        assertEquals("Application ID for call state not equal", applicationId, callStateResponse.getResult().getApplicationId());
-        assertEquals("To phone number for call state not equal", to, callStateResponse.getResult().getTo());
-        assertEquals("From phone number for call state not equal", from, callStateResponse.getResult().getFrom());
-        assertEquals("Call ID not equal", callId, callStateResponse.getResult().getCallId());
+        // String callId = createCallResponse.getResult().getCallId();
+        // ApiResponse<CallState> callStateResponse = voiceController.getCall(accountId, callId);
+        // assertEquals("Application ID for call state not equal", applicationId, callStateResponse.getResult().getApplicationId());
+        // assertEquals("To phone number for call state not equal", to, callStateResponse.getResult().getTo());
+        // assertEquals("From phone number for call state not equal", from, callStateResponse.getResult().getFrom());
+        // assertEquals("Call ID not equal", callId, callStateResponse.getResult().getCallId());
```
Comment (hamermike) [inline src/test/java/com/bandwidth/ApiTest.java:182]: I assume this is temporary and will get swapped with @adamrbaldwin-bw's tests?

**Review by hamermike — CHANGES_REQUESTED** [review]

File: src/test/java/com/bandwidth/ApiTest.java (hunk 4 above)
Comment (ckoegel) [reply src/test/java/com/bandwidth/ApiTest.java:182]: yes it is

**Review by hamermike — APPROVED** [review]

**Review by ajrice6713 — APPROVED** [review]

**Review by abaldwin-Bandwidth — APPROVED** [review]

**Pull request merged**